[ews-build.webkit.org] Update checkout based on specific remote

JonWBedard opened the pull request:
#### a38f826ee3ae1e34dc111dd3427ab87a16d355c3
<pre>
[ews-build.webkit.org] Update checkout based on specific remote
<a href="https://bugs.webkit.org/show_bug.cgi?id=242130">https://bugs.webkit.org/show_bug.cgi?id=242130</a>
&lt;rdar://problem/96162066&gt;

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(ConfigureBuild.add_pr_details): Set &apos;remote&apos; property.
(FetchBranches.run): Fetch &apos;origin&apos; and target remote, if applicable.
(UpdateWorkingDirectory):
(UpdateWorkingDirectory.run): Use raw &apos;git&apos; commands instead of &apos;update-webkit&apos;.
(UpdateWorkingDirectory.evaluateCommand): Deleted.
(CleanGitRepo.run): Remove &apos;fetch&apos; command.
(PushCommitToWebKitRepo.evaluateCommand): Add &apos;fetch&apos; step.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/252364@main">https://commits.webkit.org/252364@main</a>
</pre>

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -574,7 +574,14 @@ def add_pr_details(self):
 
-        if self.getProperty('project') != 'WebKit/WebKit':
+        project = self.getProperty('project')
+        if project == GITHUB_PROJECTS[0]:
```
This adds dependency on the order of elements in this list. If we need to do that, we should atleast add a comment where this list is defined indicating that the order matters.

JonWBedard replied:
Not the first time we've done this, that being said, the comment is a good idea.

JonWBedard committed:
[ews-build.webkit.org] Update checkout based on specific remote
https://bugs.webkit.org/show_bug.cgi?id=242130
<rdar://problem/96162066>

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(ConfigureBuild.add_pr_details): Set 'remote' property.
(FetchBranches.run): Fetch 'origin' and target remote, if applicable.
(UpdateWorkingDirectory):
(UpdateWorkingDirectory.run): Use raw 'git' commands instead of 'update-webkit'.
(UpdateWorkingDirectory.evaluateCommand): Deleted.
(CleanGitRepo.run): Remove 'fetch' command.
(PushCommitToWebKitRepo.evaluateCommand): Add 'fetch' step.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: https://commits.webkit.org/252364@main
